### PR TITLE
migrate build and test from circleci

### PR DESCRIPTION
**Pull request by lyhe18**
migrate all the tests from .circleci/config.yml. 
related to #815

### CODE REVIEW

File: .github/workflows/python-build-and-test.yml [hunk 1]
```diff
@@ -19,3 +19,33 @@ jobs:
...
+    - name: Run py.test tests excluding integration test
+      run: |
+        python3 -m pytest --ignore quarkchain/cluster/tests/test_cluster.py
+    - name: Run py.test tests only on integration test
+      run: |
+        python3 -m pytest quarkchain/cluster/tests/test_cluster.py
+    - name: Run EVM tests
+      run: |
+        python3 quarkchain/evm/tests/test_state.py
+    - name: Download DB snapshot
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: no need for line between 31 to 51. they will be a separate workflow

**Comment by ninjaahhh** [comment]
also, cancelling the workflow in case it will be stuck

File: .github/workflows/python-build-and-test.yml (hunk 1 above)
Comment (lyhe18) [reply .github/workflows/python-build-and-test.yml]: Done.

**Comment by lyhe18** [comment]
> also, cancelling the workflow in case it will be stuck

File: .github/python-build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Python unit tests
```
Comment (ninjaahhh) [inline .github/python-build-and-test.yml]: file path seems wrong

File: .github/workflows/python-build-and-test.yml (hunk 1 above)
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: as I said earlier, we don't need check-db workflow, from line 31 to end of the file

File: .github/python-build-and-test.yml (hunk 2 above)
Comment (lyhe18) [reply .github/python-build-and-test.yml]: Done.

File: .github/workflows/python-build-and-test.yml (hunk 1 above)
Comment (lyhe18) [reply .github/workflows/python-build-and-test.yml]: Done.

File: .github/workflows/python-build-and-test.yml [hunk 3]
```diff
@@ -2,20 +2,88 @@ name: Python unit tests
 
 on: [push]
 
+env:
+  PYTHONPATH: ${{ github.workspace }}
+
 jobs:
-  build:
+  build1:
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: try more explicit naming, this one is running integration test and EVM test so it's probably `integration-and-evm-test`

File: .github/workflows/python-build-and-test.yml [hunk 4]
```diff
@@ -2,20 +2,88 @@ name: Python unit tests
...
+
+    runs-on: ubuntu-latest
+    container: quarkchaindocker/pyquarkchain:mainnet1.1.1
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install Project Dependencies
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: I'll suggest combine all pre-setups together as 

```yaml
    - name: Setup - install dependencies and build qkchashlib
      env:
        PYTHONPATH: /code/pyquarkchain
      run: |
        pip3 install -e .
        cd qkchash && make clean && make
```

File: .github/workflows/python-build-and-test.yml [hunk 5]
```diff
@@ -2,20 +2,88 @@ name: Python unit tests
...
+        python3 quarkchain/evm/tests/test_state.py fixtures/QuarkChainStateTests/
+
+      
+  build2:
+    runs-on: ubuntu-latest
+    container: quarkchaindocker/pyquarkchain:mainnet1.1.1
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install Project Dependencies
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: same as previous

File: .github/workflows/python-build-and-test.yml [hunk 6]
```diff
@@ -2,20 +2,88 @@ name: Python unit tests
...
+        cd qkchash && make clean && make && cd -
+        python3 -m pytest quarkchain/cluster/tests/test_cluster.py
+
+    - name: Run QuarkChain EVM tests
+      run: |
+        cd qkchash && make clean && make && cd -
+        python3 quarkchain/evm/tests/test_state.py fixtures/QuarkChainStateTests/
+
+      
+  build2:
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: more explicit naming

File: .github/workflows/python-build-and-test.yml [hunk 7]
```diff
@@ -2,20 +2,88 @@ name: Python unit tests
...
+      run: |
+        pip3 install -e .
 
+    - name: Run py.test tests excluding integration test
+      run: |
+        cd qkchash && make clean && make && cd -
+        python3 -m pytest --ignore quarkchain/cluster/tests/test_cluster.py
+
+
+  build3:
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: naming

File: .github/workflows/python-build-and-test.yml [hunk 8]
```diff
@@ -2,20 +2,88 @@ name: Python unit tests
...
     runs-on: ubuntu-latest
     container: quarkchaindocker/pyquarkchain:mainnet1.1.1
 
     steps:
     - uses: actions/checkout@v2
-    - name: Build qkchash
+    - name: Install Project Dependencies
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: combine steps

File: .github/workflows/python-build-and-test.yml [hunk 9]
```diff
@@ -2,20 +2,88 @@ name: Python unit tests
...
+      env:
+        PYTHONPATH: /code/pyquarkchain
+      run: |
+        pip3 install -e .
+
+    - name: Run EVM tests - runner 1
       run: |
         cd qkchash && make clean && make && cd -
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: evm tests don't need qkchash lib

File: .github/workflows/python-build-and-test.yml [hunk 10]
```diff
@@ -2,20 +2,88 @@ name: Python unit tests
...
+        PYTHONPATH: /code/pyquarkchain
       run: |
         pip3 install -e .
-    - name: Test with pytest
+
+    - name: Run EVM tests - runner 2
       run: |
-        python3 -m pytest quarkchain/cluster/tests/test_cluster.py
+        cd qkchash && make clean && make && cd -
+        find fixtures/GeneralStateTests -name "*.json" | sort | sed -n 'p;n' | xargs cp --parents -t /tmp/ && \
```
Comment (ninjaahhh) [inline .github/workflows/python-build-and-test.yml]: this test doesn't need qkchash lib

**Comment by ninjaahhh** [comment]
please use a separate PR for check-db CI, and focus on commit-build-test-workflow in this one

**Pull request closed without merging**